openwillis/measures/audio/util/separation_util.py

```diff
@@ -670,23 +479,21 @@ def get_segment_signal(audio_signal, df):
 
     ------------------------------------------------------------------------------------------------------
     """
-    spk0_audio = []
-    spk1_audio = []
+    signal_dict = {}
 
     for index, row in df.iterrows():
         start_time = row['start_time']
+
         end_time = row['end_time']
         speaker_label = row['speaker_label']
 
         speaker_audio = audio_signal[start_time:end_time]
         speaker_array = np.array(speaker_audio.get_array_of_samples())
 
-        if speaker_label == 'spk_0':
-            spk0_audio.append(speaker_array)
-        elif speaker_label == 'spk_1':
-            spk1_audio.append(speaker_array)
+        if speaker_label in ['speaker0', 'speaker1', 'clinician', 'participant']:
+            signal_dict.setdefault(speaker_label, []).append(speaker_array)
 
-    return spk0_audio, spk1_audio
```

Comment:
update docstring of function for different returns

Reply:
updated